Assets/LeapMotion/Scripts/DetectionUtilities/FingerDirectionDetector.cs

```diff
@@ -0,0 +1,122 @@
+    [Tooltip("The hand model to watch. Set automatically if detector is on a hand.")]
+    public IHandModel handModel = null;  
+    public Finger.FingerType FingerName = Finger.FingerType.TYPE_INDEX;
+    [Tooltip("The target direction.")]
+    public Vector3 PointingDirection = Vector3.forward;
+    [Tooltip("How to treat the target direction.")]
+    public PointingType PointingType = PointingType.RelativeToHorizon;
+    [Tooltip("A target object(optional). Use PointingType.AtTarget")]
+    public Transform TargetObject = null;
+    [Tooltip("The angle in degrees from the target direction at which to turn on.")]
+    [Range(0, 360)]
+    public float OnAngle = 15f; //degrees
+    [Tooltip("The angle in degrees from the target direction at which to turn off.")]
+    [Range(0, 360)]
+    public float OffAngle = 25f; //degrees
+
+    private void OnValidate(){
+      if( OffAngle < OnAngle){
+        OffAngle = OnAngle;
+      }
+    }
+
+    private void Start () {
+      if(handModel == null){
+        handModel = gameObject.GetComponentInParent<IHandModel>();
+      }
+    }
+
+    private void OnEnable () {
+      StartCoroutine(fingerPointingWatcher());
+    }
+  
+    private void OnDisable () {
+      StopCoroutine(fingerPointingWatcher());
+      Deactivate();
+    }
+
+    private IEnumerator fingerPointingWatcher() {
+      Hand hand;
+      Vector3 fingerDirection;
+      Vector3 targetDirection;
+      int selectedFinger = selectedFingerOrdinal();
+      while(true){
+        if(handModel != null){
+          hand = handModel.GetLeapHand();
+          if(hand != null){
+            targetDirection = selectedDirection(hand.Fingers[selectedFinger].TipPosition.ToVector3());
+            fingerDirection = hand.Fingers[selectedFinger].Direction.ToVector3();
+            float angleTo = Vector3.Angle(fingerDirection, targetDirection);
+            if(handModel.IsTracked && angleTo <= OnAngle){
+              Activate();
+            } else if (!handModel.IsTracked || angleTo >= OffAngle) {
+              Deactivate();
+            }
+          }
+        }
+        yield return new WaitForSeconds(Period);
+      }
+    }
+
+    private Vector3 selectedDirection(Vector3 tipPosition){
+      switch(PointingType){
+        case PointingType.RelativeToHorizon:
+          Quaternion cameraRot = Camera.main.transform.rotation;
+          float cameraYaw = cameraRot.eulerAngles.y;
+          Quaternion rotator = Quaternion.AngleAxis(cameraYaw, Vector3.up);
+          return rotator * PointingDirection;
+        case PointingType.RelativeToCamera:
+          return Camera.main.transform.TransformDirection(PointingDirection);
+        case PointingType.RelativeToWorld:
+          return PointingDirection;
+        case PointingType.AtTarget:
+          return TargetObject.position - tipPosition;
+        default:
+          return PointingDirection;
+      }
+    }
+
+    private int selectedFingerOrdinal(){
```

Comment:
Yeah I am proposing that we add a method to Hand that takes in a finger type and returns the finger of that type (or null / error if you pass in type unknown).  The hand class could use the enum value as an index, just with some extra checks for unknown enum types.  That way this functionality (which is quite useful in all cases!) can be used all over the place.